Dockerfile

```diff
@@ -12,43 +12,34 @@
 # 
 # docker build  --build-arg BUILD_REPO=YourFork/PokemonGo-Bot --build-arg BUILD_BRANCH=6a4580f .
 
-FROM python:2.7.12-alpine
+FROM alpine
 
-RUN apk add --update --no-cache alpine-sdk git
-
-ARG BUILD_BRANCH
-ENV BUILD_BRANCH ${BUILD_BRANCH:-master}
-
-ARG BUILD_REPO
-ENV BUILD_REPO ${BUILD_REPO:-PokemonGoF/PokemonGo-Bot}
+ARG BUILD_REPO=PokemonGoF/PokemonGo-Bot
+ARG BUILD_BRANCH=master
+ARG TIMEZONE=Etc/UTC
 
 LABEL build_repo=$BUILD_REPO build_branch=$BUILD_BRANCH
 
-ADD https://github.com/$BUILD_REPO/archive/$BUILD_BRANCH.tar.gz .
-RUN tar -zxvf $BUILD_BRANCH.tar.gz && mv PokemonGo-Bot-* /usr/src/app && rm $BUILD_BRANCH.tar.gz
-
 WORKDIR /usr/src/app
 VOLUME ["/usr/src/app/configs", "/usr/src/app/web"]
 
-ARG timezone=Etc/UTC
-RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone
-
-#setup the bot
-ADD http://pgoapi.com/pgoencrypt.tar.gz /tmp/
-RUN cd /tmp && tar zxvf pgoencrypt.tar.gz \
-    && cd pgoencrypt/src \
-    && make \
-    && cp libencrypt.so /usr/src/app/encrypt.so \
-    && cd /tmp \
-    && rm -rf /tmp/pgoencrypt*
-
-ENV LD_LIBRARY_PATH /usr/src/app
-
-RUN ln -s /usr/include/locale.h /usr/include/xlocale.h
-RUN pip install --no-cache-dir -r requirements.txt
-
-#remove unused stuff
-RUN apk del alpine-sdk\
-  && rm -rf /var/cache/apk/*
+ADD https://raw.githubusercontent.com/$BUILD_REPO/$BUILD_BRANCH/requirements.txt .
+RUN apk -U --no-cache add python py-pip \
+    && apk --no-cache add --virtual .build-dependencies python-dev gcc make musl-dev git tzdata \
+    && cp -fa /usr/share/zoneinfo/$TIMEZONE /etc/localtime \
+    && echo $TIMEZONE > /etc/timezone \
+    && wget -q -O- http://pgoapi.com/pgoencrypt.tar.gz | tar zxf - -C /tmp \
+    && make -C /tmp/pgoencrypt/src \
+    && cp /tmp/pgoencrypt/src/libencrypt.so /usr/src/app/encrypt.so \
+    && ln -s locale.h /usr/include/xlocale.h \
+    && pip install --no-cache-dir -r requirements.txt \
+    && apk del .build-dependencies \
+    && rm -rf /var/cache/apk/* /tmp/pgoencrypt /usr/include/xlocale.h \
+    && find / -name '*.pyc' -o -name '*.pyo' -exec rm -f {} \;
+
+RUN apk -U --no-cache add --virtual .install-dependencies wget ca-certificates tar \
+    && wget -q -O- https://github.com/$BUILD_REPO/archive/$BUILD_BRANCH.tar.gz | tar zxf - --strip-components=1 -C /usr/src/app \
```

Comment:
yes, and more quickly!

it alse cached requirements.txt layer.
if this file no change, it just download the latest src to image and create new image layer.